temporalio/client.py

```diff
@@ -1165,29 +1165,68 @@ async def report_cancellation(self, *details: Any) -> None:
         )
 
 
-class WorkflowDescription:
-    """Description for a workflow."""
-
-    def __init__(
-        self,
-        raw_message: temporalio.api.workflowservice.v1.DescribeWorkflowExecutionResponse,
-    ):
-        """Create a workflow description from a describe response."""
-        self._raw_message = raw_message
-        status = raw_message.workflow_execution_info.status
-        self._status = WorkflowExecutionStatus(status) if status else None
+@dataclass
+class WorkflowExecutionDescription:
+    """Description for a workflow execution."""
 
-    @property
-    def raw_message(
-        self,
-    ) -> temporalio.api.workflowservice.v1.DescribeWorkflowExecutionResponse:
-        """Underlying workflow description response."""
-        return self._raw_message
+    close_time: Optional[datetime]
+    execution_time: Optional[datetime]
+    history_length: int
+    id: str
+    memo: Mapping[str, Any]
+    parent_id: Optional[str]
+    parent_run_id: Optional[str]
+    raw: temporalio.api.workflowservice.v1.DescribeWorkflowExecutionResponse
+    run_id: str
+    search_attributes: temporalio.common.SearchAttributes
+    start_time: datetime
+    status: Optional[WorkflowExecutionStatus]
+    task_queue: str
+    workflow: str
```

Comment:
I'd call this workflow_name or workflow_type or just type if it wasn't a built-in.

Reply:
Due to the ambiguity between "type", "name", "type name", etc, I have been using `workflow` or `activity` in most places. This is the case in interceptors, in the parameter names of `start_workflow` or `execute_activity`, etc. Granted for workflow/activity info, I still use `workflow_type`/`activity_type`, so I guess I can change it here.